tty: add hasColors function

BridgeAR opened the pull request:
This adds a small wrapper around the `getColorDepth` function to check
if the stream supports at least a specific amount of colors. This is
convenient as the other API is not as straight forward and most use
cases likely only want to know if a specific amount of colors is
supported or not.

<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

addaleax commented on doc/api/tty.md:
```diff
@@ -176,6 +176,34 @@ corresponding to this `WriteStream`. The array is of the type
+* Returns: {boolean}
+
+Returns true if the writeStream supports at least as many colors as provided in
+`count`. If the stream only supports black and white (the minimum), it'll
```
`it'll support` → `it [only] supports`?

BridgeAR commented:
CI https://ci.nodejs.org/job/node-test-pull-request/20952/

BridgeAR commented:
Resumed CI https://ci.nodejs.org/job/node-test-commit/26074/

vsemozhetbyt commented on doc/api/tty.md:
```diff
@@ -176,6 +176,34 @@ corresponding to this `WriteStream`. The array is of the type
+  **Default:** `process.env`.
+* Returns: {boolean}
+
+Returns true if the writeStream supports at least as many colors as provided in
```
Nit with rewpapping:
```suggestion
Returns `true` if the `writeStream` supports at least as many colors as provided in
```

jasnell commented on doc/api/tty.md:
```diff
@@ -176,6 +176,34 @@ corresponding to this `WriteStream`. The array is of the type
+* `count` {integer} The number of colors that are requested (minimum 2).
+  **Default:** 16.
+* `env` {Object} An object containing the environment variables to check.
+  **Default:** `process.env`.
```
Just a nit... it's not clear from the documentation how the `env` bit works here. Perhaps an example?

BridgeAR replied:
There are two examples below but I'll try to improve the wording further.

BridgeAR replied:
PTAL

BridgeAR commented:
CI https://ci.nodejs.org/job/node-test-pull-request/21047/

BridgeAR commented:
@jasnell PTAL

BridgeAR commented:
I am not sure why but Github reported conflicts, so I rebased once more. Locally there was no conflict and I just squashed the commits, no other change.

CI https://ci.nodejs.org/job/node-test-pull-request/21210/

BridgeAR commented:
Landed in d4fdec6b6593659cb38e1db9c8083d4fde9bcacb :tada:
